### PR TITLE
Patch the graph_type creation bug

### LINKED ISSUE #3236: Graph type is null on graph creation
Screenshot_2024-06-20_09-41-27

It appears the graph type needs to be passed in to the form somehow. This is a React form, so there are many ways to do this. One possible way is the following:

We can make use of the unused `type` variable in the `CohortGraphCreateView`: `path('course/<int:pk>/graph/create/<int:type>/',`.
I would rename this variable to `graph_type`, because `type` is a keyword in Python. From this view, we can send graph_type to the template via ctx.update(). Then, we can use this to populate a new variable in the global `EconPlayground` JS object - something like this:
```
window.EconPlayground = {
   // ...
   graphType: {{ graph_type }}
   // ...
}
```

### PR DESCRIPTION
**Pull request by Evan-CTL**
Closes #3236

### CODE REVIEW

File: econplayground/templates/main/graph_form.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 {% block js %}
     <script src="{% static 'js/build/editor.js' %}"></script>
+    window.EconPlayground = { graphType: {{graph_type}}}
```
Comment (nikolas) [inline econplayground/templates/main/graph_form.html:14]: So this variable is already defined in base.html and we don't want to overwrite it here.
https://github.com/ccnmtl/econplayground/blob/master/econplayground/templates/base.html#L175

We can add a new variable to this object, and we'll want to check that it actually exists:
```
        var EconPlayground = {
            // All the existing data here

            {% if graph_type is not None %}
                graphType: {{graph_type}},
            {% endif %}
        };
```

Note that you'll also need to add commas to the `isStaff` variable declarations above in order for this to parse.

Comment (nikolas) [reply econplayground/templates/main/graph_form.html:14]: Or, we can conditionally declare this object within this template, and just add this new variable to it within this template - following this pattern: https://github.com/ccnmtl/econplayground/blob/master/econplayground/templates/main/graph_detail.html#L18

Maybe that is a bit cleaner as this variable is only present for this specific view. But functionally, it doesn't matter so much either way.

Comment (Evan-CTL) [reply econplayground/templates/main/graph_form.html]: I went for the implementation that wouldn't affect the base as this variable is only necessary in graph creation.

File: econplayground/templates/main/graph_form.html [hunk 2]
```diff
@@ -10,6 +10,10 @@
 {% endblock %}
 
 {% block js %}
+    if (!window.EconPlayground) {
+        window.EconPlayground = {};
+    }
+    window.Econplayground[graphType] =  {{graph_type}}
```
Comment (nikolas) [inline econplayground/templates/main/graph_form.html:16]: This block of code needs to be surrounded by a `<script>` tag:
```
<script>
    if (!window.EconPlayground) {
        window.EconPlayground = {};
    }
    window.EconPlayground[graphType] =  {{graph_type}};   // <-- also add semicolon to this line
    // and fix typo in EconPlayground variable here, capitalizing the P.
</script>
```

Comment (Evan-CTL) [reply econplayground/templates/main/graph_form.html:16]: Okay will do. Why is the js block not already in a script tag?

**Pull request merged**